### PR TITLE
Set default pool size for HiLo to 1 and allow pool size to be configured

### PR DESCRIPTION
**Pull request by ajcvickers**
As per design meeting decision and Issue #778. Also some renaming in core metadata to better match the new naming in the fluent API.

### CODE REVIEW

File: src/EntityFramework.SqlServer/Metadata/SqlServerAnnotationNames.cs [hunk 1]
```diff
@@ -9,9 +9,8 @@ public static class SqlServerAnnotationNames
         public const string Clustered = "Clustered";
         public const string ItentityStrategy = "ItentityStrategy";
```
Comment (smitpatel) [inline src/EntityFramework.SqlServer/Metadata/SqlServerAnnotationNames.cs]: Good time to update this also.
see #2524

Comment (Artur2) [inline src/EntityFramework.SqlServer/Metadata/SqlServerAnnotationNames.cs]: Identity ?

**Comment by ajcvickers** [comment]
@smitpatel Will do

**Comment by ajcvickers** [comment]
@smitpatel Are you planning to finish review of this? If not, I'll find somebody else to look at it.

File: src/EntityFramework.SqlServer/SqlServerModelBuilderExtensions.cs [hunk 2]
```diff
@@ -74,8 +95,8 @@ public static class SqlServerModelBuilderExtensions
             var property = modelBuilder.Model;
 
             property.SqlServer().IdentityStrategy = SqlServerIdentityStrategy.IdentityColumn;
-            property.SqlServer().DefaultSequenceName = null;
-            property.SqlServer().DefaultSequenceSchema = null;
+            property.SqlServer().HiLoSequenceName = null;
+            property.SqlServer().HiLoSequenceSchema = null;
```
Comment (smitpatel) [inline src/EntityFramework.SqlServer/SqlServerModelBuilderExtensions.cs]: Pool Size also null/default?

File: src/EntityFramework.SqlServer/SqlServerPropertyBuilderExtensions.cs [hunk 3]
```diff
@@ -132,8 +161,8 @@ public static class SqlServerPropertyBuilderExtensions
 
             property.SqlServer().IdentityStrategy = SqlServerIdentityStrategy.IdentityColumn;
             property.ValueGenerated = ValueGenerated.OnAdd;
-            property.SqlServer().SequenceName = null;
-            property.SqlServer().SequenceSchema = null;
+            property.SqlServer().HiLoSequenceName = null;
+            property.SqlServer().HiLoSequenceSchema = null;
```
Comment (smitpatel) [inline src/EntityFramework.SqlServer/SqlServerPropertyBuilderExtensions.cs]: PoolSize?

File: src/EntityFramework.SqlServer/SqlServerModelBuilderExtensions.cs (hunk 2 above)
Comment (ajcvickers) [inline src/EntityFramework.SqlServer/SqlServerModelBuilderExtensions.cs]: Will do.

File: src/EntityFramework.SqlServer/SqlServerPropertyBuilderExtensions.cs (hunk 3 above)
Comment (ajcvickers) [inline src/EntityFramework.SqlServer/SqlServerPropertyBuilderExtensions.cs]: Will do.

File: test/EntityFramework.SqlServer.Tests/Metadata/SqlServerBuilderExtensionsTest.cs [hunk 4]
```diff
@@ -15,7 +15,7 @@ namespace Microsoft.Data.Entity.SqlServer.Tests.Metadata
     public class SqlServerBuilderExtensionsTest
```
Comment (smitpatel) [inline test/EntityFramework.SqlServer.Tests/Metadata/SqlServerBuilderExtensionsTest.cs]: ValidateDefaultSequence, ValidateDefaultSpecificSequence dead code in this file. possibly safe to remove.

**Comment by smitpatel** [comment]
:shipit:

File: test/EntityFramework.SqlServer.Tests/Metadata/SqlServerBuilderExtensionsTest.cs (hunk 4 above)
Comment (ajcvickers) [inline test/EntityFramework.SqlServer.Tests/Metadata/SqlServerBuilderExtensionsTest.cs]: Already removed in a future pull request.

**Comment by ajcvickers** [comment]
Merged

**Pull request closed without merging**